app/app/explore/components/ExploreTable.tsx

```diff
@@ -0,0 +1,359 @@
+import { useState, useMemo, useEffect } from "react";
+import Image from "next/image";
+import TranslucentCard from "@/app/TranslucentCard";
+import TopCards from "@/app/explore/TopCards";
+import { DoubleArrow } from "@shared/components/svg";
+import { PoolDetail, PoolDetails } from "@/midgard";
+import {
+  addDollarSignAndSuffix,
+  calculatePoolTVL,
+  calculateVolumeDepthRatio,
+  calculateVolumeUSD,
+  calculateSaverTVL,
+  formatNumber,
+  getFormattedPoolTVL,
+  getFormattedSaverTVL,
+} from "@/app/utils";
+import SortHeader, { SortKey } from "./SortHeader";
+
+interface Saver {
+  asset: string;
+  saversCount: number;
+  saversReturn: string;
+  earned: string;
+  filled: number;
+  assetPriceUSD: string;
+  saversDepth: string;
+  assetDepth: string;
+  synthSupply: string;
+}
+
+enum SortDirection {
+  ASC = "asc",
+  DESC = "desc",
+}
+
+interface SortConfig {
+  key: SortKey;
+  direction: SortDirection;
+}
+
+interface Column {
+  key: string;
+  label: string;
+  width: string;
+  sortable?: boolean;
+  render: (item: any) => React.ReactNode;
+}
+
+interface ExploreTableProps {
+  type: "pools" | "savers";
+  data: PoolDetails | Saver[];
+  runePriceUSD?: number;
+}
+
+const getAssetSymbol = (asset: string) => {
+  return asset.split("-")[0] || asset;
+};
+
+const getLogoPath = (asset: string) => {
+  const assetLower = asset.toLowerCase();
+  return `https://storage.googleapis.com/token-list-swapkit-dev/images/${assetLower}.png`;
+};
+
+export default function ExploreTable({
+  type,
+  data,
+  runePriceUSD,
+}: ExploreTableProps) {
+  const [isMobile, setIsMobile] = useState(false);
+  const [sortConfig, setSortConfig] = useState<SortConfig>({
+    key: SortKey.TVL,
+    direction: SortDirection.DESC,
+  });
+
+  useEffect(() => {
+    const handleResize = () => {
+      setIsMobile(window.innerWidth < 768);
+    };
+
+    handleResize();
+    window.addEventListener("resize", handleResize);
+    return () => window.removeEventListener("resize", handleResize);
+  }, []);
+
+  const columns: Column[] =
+    type === "pools"
```

Comment:
i think it's bad design to have to check for `type` everywhere in order to accommodate pools and savers. it's fine to have ExploreTable accept a `type` prop, but there should be independent components for Savers and Pools. There's a lot of shared data like `isMobile` and `sortConfig`, so they should passed into the child components.